test/elixometer_test.exs

```diff
@@ -268,6 +271,47 @@ defmodule ElixometerTest do
     assert get_metric_value("elixometer.test.spirals.precomputed_counter", :one) == {:ok, 123}
   end
 
+  test "name can be precomputed with env config from environmental variable" do
+    System.put_env("ELIXOMETER_ENV", "staging")
+    original_env = Application.get_env(:elixometer, :env)
+    Application.put_env(:elixometer, :env, {:system, "ELIXOMETER_ENV"})
+
+    name = Elixometer.Utils.name_to_exometer(:spirals, "precomputed_counter")
+    update_spiral(name, 123)
+    wait_for_messages()
+
+    assert get_metric_value("elixometer.staging.spirals.precomputed_counter", :one) == {:ok, 123}
+
+    Application.put_env(:elixometer, :env, original_env)
+  end
+
+  test ~s(name can be precomputed with env config from environmental variable and env is "prod") do
+    System.put_env("ELIXOMETER_ENV", "prod")
+    original_env = Application.get_env(:elixometer, :env)
+    Application.put_env(:elixometer, :env, {:system, "ELIXOMETER_ENV"})
+
+    name = Elixometer.Utils.name_to_exometer(:spirals, "precomputed_counter")
+    update_spiral(name, 123)
+    wait_for_messages()
+
+    assert get_metric_value("elixometer.spirals.precomputed_counter", :one) == {:ok, 123}
+
+    Application.put_env(:elixometer, :env, original_env)
```

Comment:
Since it's in the `on_exit` callback it's no longer needed in the tests. Removing ✂️ .